### ISSUE TITLE
[web]: Fail to announce button press when using enter key

### ISSUE THREAD

**Issue by mariamhas** [flutter/flutter#83812]
**issue:** Screenreader is not announcing what button is pressed (using enter key). Also related, the snack bar that appears when a button is pressed isn't announced or read out loud.

Expectation: Screenreader should indicate that the button is pressed by announcing the action. Snackbar announcement should trigger when snackbar pops up. 

Tested using JAWs, NVDA, VoiceOver on Chrome

Sample code: https://github.com/morrison999/aira-flutter

**Comment by markusaksli-nc** [comment]
Reproducible on the latest master `2.3.0-17.0.pre.180` with Narrator.
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 2.3.0-17.0.pre.180, on Microsoft Windows [Version 10.0.19041.985], locale et-EE)
    • Flutter version 2.3.0-17.0.pre.180 at C:\Development\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 320f27713b (6 hours ago), 2021-06-03 02:49:03 -0400
    • Engine revision fe6d19b911
    • Dart version 2.14.0 (build 2.14.0-176.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\202.7351085\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.2.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\202.7351085
    • Flutter plugin version 57.0.2
    • Dart plugin version 202.8488
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[√] IntelliJ IDEA Ultimate Edition (version 2021.1)
    • IntelliJ at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\211.7442.40
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19041.985]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 91.0.4472.77
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 91.0.864.37

• No issues found!
```
</details>

**Comment by yjbanov** [comment]
This issue has the same cause as https://github.com/flutter/flutter/issues/83809. Deduplicating
